src/librustc_ty/instance.rs

```diff
@@ -67,12 +67,19 @@ fn inner_resolve_instance<'tcx>(
                 let ty = substs.type_at(0);
 
                 if ty.needs_drop(tcx, param_env) {
-                    // `DropGlue` requires a monomorphic aka concrete type.
-                    if ty.needs_subst() {
-                        return Ok(None);
+                    debug!(" => nontrivial drop glue");
+                    match ty.kind {
+                        ty::Closure(..)
+                        | ty::Generator(..)
+                        | ty::Tuple(..)
+                        | ty::Adt(..)
+                        | ty::Dynamic(..)
+                        | ty::Array(..)
+                        | ty::Slice(..) => (),
```

Comment:
*Very* minor nit: I prefer `=> {}` (but I don't know that there's an official style).

Reply:
Style RFC exists (https://github.com/rust-dev-tools/fmt-rfcs/issues/146) but has not been merged yet, your preference seems to be the popular choice though.

Reply:
Fixed!